packages/react/src/Button/Button.test.tsx

```diff
@@ -48,16 +48,76 @@ describe('Button', () => {
     expect(btnEl.classList).toContain(disabledClass)
   })
 
-  it('renders aria-disabled state', async () => {
+  it('sets the disabled attribute to true when disabled prop is true and component is rendered as a button', () => {
     const {getByRole} = render(
-      <Button variant="primary" size="medium" aria-disabled="true">
+      <Button variant="primary" size="medium" disabled>
         Primary Button
       </Button>,
     )
     const btnEl = getByRole('button')
 
-    expect(btnEl).toHaveAttribute('aria-disabled')
-    expect(btnEl.classList).toContain(disabledClass)
+    expect(btnEl).toHaveAttribute('disabled')
+    expect(btnEl).not.toHaveAttribute('aria-disabled')
+  })
+
+  it('sets the aria-disabled attribute to true when disabled prop is true and component is rendered as a link', () => {
+    const {getByText} = render(
+      <Button variant="primary" size="medium" disabled as="a">
+        Primary Button
+      </Button>,
+    )
+    const linkEl = getByText('Primary Button').closest('a')
```

Comment:
Do we need to use `closest` here? Can we use `getByRole` instead like we do for the other stories?

Could the reason for the matcher not working be that you're missing the href here?

```suggestion
    const { getByRole } = render(
      <Button as="a" href="#" disabled>
        Primary Button
      </Button>,
    );
    const linkEl = getByRole('link');
```

Removed the other unrelated props to keep the tests clean. Can we do same to others too please.